pkg/terra/config.go

```diff
@@ -60,16 +96,16 @@ type ConfigSet struct {
 var _ Config = (*config)(nil)
 
 type config struct {
-	defaults ConfigSet
+	defaults configSet
 	chain    db.ChainCfg
 	chainMu  sync.RWMutex
 	lggr     Logger
 }
 
-// NewConfig returns a Config with defaults overridden by dbcfg.
-func NewConfig(dbcfg db.ChainCfg, defaults ConfigSet, lggr Logger) *config {
+// NewConfig returns a Config with defaults for id overridden by dbcfg.
+func NewConfig(id string, dbcfg db.ChainCfg, lggr Logger) *config {
```

Comment:
```suggestion
func NewConfig(chainID string, dbcfg db.ChainCfg, lggr Logger) *config {
```

Reply:
Or perhaps even type it explicitly as `id ChainID`, then move the `defaultsForID` method to the new `ChainID` type as `defaults`.

Reply:
I like the clarity of the helper, but using it for the parameter type is a bit of a burden for the callers since conversion has to be explicit.

Reply:
Yup, crossed my mind. I'm good either way for now. Consider doing what feels better to you.